src/ser/raw/document_serializer.rs

```diff
@@ -225,7 +226,10 @@ impl<'a> serde::Serializer for KeySerializer<'a> {
 
     #[inline]
     fn serialize_u64(self, v: u64) -> Result<Self::Ok> {
-        Err(Self::invalid_key(v))
+        use std::io::Write;
```

Comment:
should we be concerned at all about users accidentally encountering this and being surprised by the behavior, e.g. if they are working with a HashMap<i64, String> and try to serialize that to BSON?